src/System.Diagnostics.Process/src/System/Diagnostics/ProcessStartInfo.cs

```diff
@@ -93,9 +93,18 @@ public bool CreateNoWindow
                         CaseSensitiveEnvironmentVariables ? StringComparer.Ordinal : StringComparer.OrdinalIgnoreCase);
 #pragma warning restore 0429
 
-                    foreach (DictionaryEntry entry in envVars)
+                    IDictionaryEnumerator e = envVars.GetEnumerator();
```

Comment:
I'd suggest adding a short comment as to why this isn't using foreach.  Otherwise, I wouldn't be surprised if someone in the future tried to "simplify" it back.